### PR TITLE
Config reintro

### PR DESCRIPTION
**Pull request by DGuhr**
Draft PR to introduce configuration via viper and a yaml file (for now).

**General Workflow:**
1) config is loaded in app.go using the interface inside `serviceconfig.Config`.  Technical impl for now is [viper](https://github.com/spf13/viper). Doing it this way allows to switch from viper to another one if we want to. Inspired by https://github.com/aellwein/config/
2) untyped, unvalidated viper-loaded config is then parsed into (renamed & moved) `serviceconfig.ServiceConfig.go` & validated at bootstrap time (using [go-validator](https://github.com/go-playground/validator), implemented only an example for the ports for now)
3) The typed and validated config is then made available in a global variable (though I'd like to avoid this, and instead initialize the services/repositories using a reference to the struct)

**Technical:**
- Tests altered to use mockConfig. Tried to create an in-memory file using [afero](https://github.com/spf13/afero), but it wasn't straight forward, so stopped for now.
- minor changes to makefile to give more convenient output
- removed the log workaround for glog, as it is not needed anymore it seems. Got same output with and without it (but: weird start / ending chars around the date, see screenshot)

Screenshot of failing server startup + logs (using fatal for now, so stacktrace is added, maybe we don't want that):Screenshot 2023-04-26 at 09 05 42

### **TODO:**
- first of all, decide we want to go further down this road and do the necessary paperwork ;p. 
- if so, open things I see are: 
  * ~Integrate config into middleware chain.~  Already integrated, thx to @merlante for providing an easy solution in #118 👍 
  * Add more validations (and maybe better error messages)
  * change Kind & pipelines, as this is a breaking change.

### PR Template:

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: bootstrap/app.go [hunk 1]
```diff
@@ -17,9 +19,42 @@ var grpcServer *grpc.Server
...
+	err = vl.Struct(srvCfg)
+
+	if err != nil {
+		for _, e := range err.(validator.ValidationErrors) {
+			glog.Errorf("Error in configuration: %v", e)
+		}
+		glog.Fatal("Can not start service with wrong configuration.")
+	}
+	//set global Config now that it is parsed and validated.
+	Cfg = &srvCfg
```
Comment (merlante) [inline bootstrap/app.go:55]: Hmm... no problem with global config, but shouldn't we then access config globally and not pass srvCfg everywhere as an argument?

I suggest we remove this until we are ready/willing to use it.

File: bootstrap/serviceconfig/ServiceConfig.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+const (
+	// SpicedbImage is the image used for containerized spiceDB in tests
+	SpicedbImage = "authzed/spicedb"
+	// SpicedbVersion is the image version used for containerized spiceDB in tests
+	SpicedbVersion = "v1.20.0"
+)
+
+// ServiceConfig contains all server-related configuration.
+type ServiceConfig struct {
```
Comment (Rajagopalan-Ranganathan) [inline bootstrap/serviceconfig/ServiceConfig.go:12]: loglevel should be included here? WDYT?

File: infrastructure/config/ViperConfig.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// Load loads the current configuration
+func (c *ViperConfig) Load() (serviceconfig.ServiceConfig, error) {
+	err := c.v.ReadInConfig()
+	if err != nil {
+		return c.defaults, err
+	}
+
+	cfg := c.defaults //Value-assignment to copy the defaults into a new object
+	err = c.v.Unmarshal(&cfg)
+	fmt.Printf("Loaded SpiceDB preshared key: %s", cfg.StoreConfig.AuthToken)
```
Comment (Rajagopalan-Ranganathan) [inline infrastructure/config/ViperConfig.go:26]: this needs to be removed :)

File: bootstrap/app.go [hunk 4]
```diff
@@ -17,9 +19,42 @@ var grpcServer *grpc.Server
...
+		glog.Fatal("Unable to load configuration: %v", err)
+	}
+	vl := validator.New()
+	err = vl.Struct(srvCfg)
+
+	if err != nil {
+		for _, e := range err.(validator.ValidationErrors) {
+			glog.Errorf("Error in configuration: %v", e)
+		}
+		glog.Fatal("Can not start service with wrong configuration.")
```
Comment (Rajagopalan-Ranganathan) [inline bootstrap/app.go:52]: text change - "wrong" --> "invalid". WDYT?

File: bootstrap/app.go (hunk 1 above)
Comment (wscalf) [reply bootstrap/app.go:55]: This is a good point. One of the items on Dominik's big list of suggestions was to see if we could remove that- we probably can.

File: infrastructure/config/ViperConfig.go (hunk 3 above)
Comment (wscalf) [reply infrastructure/config/ViperConfig.go]: Yeeeeaaaaah it does. Removed.

File: bootstrap/serviceconfig/ServiceConfig.go (hunk 2 above)
Comment (Rajagopalan-Ranganathan) [reply bootstrap/serviceconfig/ServiceConfig.go:12]: Agreed to add this later as a followup item.

File: bootstrap/app.go (hunk 4 above)
Comment (Rajagopalan-Ranganathan) [reply bootstrap/app.go:49]: resolved.

**Comment by merlante** [comment]
Fingers crossed, here goes nothing... :)

**Pull request merged**